lib/web/apps.go

```diff
@@ -66,15 +66,58 @@ type CreateAppSessionRequest struct {
 	ClusterName string `json:"cluster_name"`
 }
 
+type GetAppFQDNResponse struct {
+	// FQDN is application FQDN.
+	FQDN string `json:"fqdn"`
+}
+
 type CreateAppSessionResponse struct {
 	// CookieValue is the application session cookie value.
 	CookieValue string `json:"value"`
 	// FQDN is application FQDN.
 	FQDN string `json:"fqdn"`
 }
 
+// getAppFQDN resolves the input params to a known application and returns
+// its valid FQDN.
+//
+// GET /v1/webapi/apps/:fqdn/:clusterName/:publicAddr
+func (h *Handler) getAppFQDN(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext) (interface{}, error) {
+	req := &AppResolveParams{
+		FQDNHint:    p.ByName("fqdnHint"),
```

Comment:
Check that fqdn hint is not empty, or will it be checked by one of resolveXXX functions below?